src/apps/front-office/design-system/layouts/Header/components/HeaderIcons/HeaderSearch/HeaderSearchFilter/HeaderSearchFilter.tsx

```diff
@@ -0,0 +1,46 @@
+import { trans } from "@mongez/localization";
+import { Link } from "@mongez/react-router";
+import URLS from "apps/front-office/utils/urls";
+import {
+  searchInputAtom,
+  toggleGroupAtom,
+} from "../../../../atoms/header-atoms";
+import useSearchMeals from "./useSearchMeals/useSearchMeals";
+
+export default function HeaderSearchFilter() {
+  const searchInput = searchInputAtom.useValue();
+  const isSearchOpened = toggleGroupAtom.useValue().headerSearch;
+
+  const { meals, error } = useSearchMeals(searchInput, isSearchOpened);
+
+  if (error) {
+    return <div>{error}</div>;
+  }
+
+  if (meals()?.length === 0) {
+    return (
+      <h1 className="bg-white w-full absolute left-0 top-[100px] p-5">
+        {trans("noResults")}
+      </h1>
+    );
+  }
+
+  return (
+    <div className="bg-white w-full absolute left-0 top-[100px] px-5 max-h-[67vh] overflow-auto rounded-md">
+      {meals()?.map(meal => (
```

Comment:
Return meals as is, it should be an array not a function, the filter should be done through API not in the browser.